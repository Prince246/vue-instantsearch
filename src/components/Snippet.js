import { getPropertyByPath } from '../util/object';
export default {
  functional: true,
  props: {
    result: {
      type: Object,
      required: true,
    },
    attributeName: {
      type: String,
      required: true,
    },
  },
  render(h, ctx) {
    const result = ctx.props.result;
    const attributeName = ctx.props.attributeName;

    const attributePath = `_snippetResult.${attributeName}.value`;
    const attributeValue = getPropertyByPath(result, attributePath);

    if (process.env.NODE_ENV !== 'production' && attributeValue === undefined) {
      throw new Error(
        `The "${attributeName}" attribute is currently not configured to be snippeted in Algolia.
        See https://www.algolia.com/doc/api-reference/api-parameters/attributesToSnippet/.`
      );
    }

    return h('span', {
      class: {
        'ais-Snippet': true,
      },
      domProps: {
        innerHTML: attributeValue,
      },
    });
  },
};
