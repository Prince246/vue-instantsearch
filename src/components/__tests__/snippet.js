import { mount } from '@vue/test-utils';
import Snippet from '../Snippet';

afterEach(() => {
  process.env.NODE_ENV = 'test';
});

test('renders proper HTML', () => {
  const result = {
    _snippetResult: {
      attr: {
        value: `con<em>ten</em>t`,
      },
    },
  };

  const wrapper = mount(Snippet, {
    context: {
      props: {
        attributeName: 'attr',
        result,
      },
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render an empty string in production if attribute is not snippeted', () => {
  process.env.NODE_ENV = 'production';
  const result = {
    _snippetResult: {},
  };
  global.console.warn = jest.fn();

  const wrapper = mount(Snippet, {
    context: {
      props: {
        attributeName: 'attr',
        result,
      },
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
  expect(global.console.warn).not.toHaveBeenCalled();
});

test('should warn when not in production if attribute is not snippeted', () => {
  global.console.warn = jest.fn();

  const result = {
    _snippetResult: {},
  };
  mount(Snippet, {
    context: {
      props: {
        attributeName: 'attr',
        result,
      },
    },
  });

  expect(global.console.warn).toHaveBeenCalledTimes(1);
});

test('allows usage of dot delimited path to access nested attribute', () => {
  const result = {
    _snippetResult: {
      attr: {
        nested: {
          value: `nested <em>val</em>`,
        },
      },
    },
  };

  const wrapper = mount(Snippet, {
    context: {
      props: {
        attributeName: 'attr.nested',
        result,
      },
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});
