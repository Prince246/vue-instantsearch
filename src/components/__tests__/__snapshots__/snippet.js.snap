// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippet allows usage of dot delimited path to access nested attribute 1`] = `

<span class="ais-Snippet">
  nested
  <em>
    val
  </em>
</span>

`;

exports[`Snippet renders proper HTML 1`] = `

<span class="ais-Snippet">
  con
  <em>
    ten
  </em>
  t
</span>

`;

exports[`Snippet should render an empty string in production if attribute is not snippeted 1`] = `

<span class="ais-Snippet">
</span>

`;
